markdown.go

```diff
@@ -218,7 +218,8 @@ type parser struct {
 	// Footnotes need to be ordered as well as available to quickly check for
 	// presence. If a ref is also a footnote, it's stored both in refs and here
 	// in notes. Slice is nil if footnotes not enabled.
-	notes []*reference
+	notes       []*reference
+	notesRecord map[string]bool
```

Comment:
`map[string]struct{}` is a better way to represent a set. It avoids allocating memory for the value.

But it also looks like we already have a map that could be used for this: `p.refs`. Can you check if it could do the job?